File: javascript/string/434. Number of Segments in a String.js
/**
 * @param {string} s
 * @return {number}
 */
// own
var countSegments = function(s) {
    if (s.length == 0) return 0;

    let count = 0;
    for (let i=1; i<s.length; i++) {
        if (s[i] == " " && s[i-1] != " ") count++;
    }
    return s[s.length-1] == " " ? count : count + 1;
};

// modification
var countSegments = function(s) {
    let count = 0;
    for (let i=0; i<s.length; i++) {
        if ((i == 0 || s[i-1] == " ") && s[i] != " ") count++;
    }
    return count;
};

File: python/string/434. Number of Segments in a String.py
'''
own
Time O(n)
Space: O(1)
elegant way
Time: O(n)
Space: O(n)
'''

class Solution:
    def countSegments(self, s: str) -> int:
        if len(s) == 0:
            return 0

        count = 0
        for i in range(1, len(s)):
            if s[i] == " " and s[i-1] != " ":
                count += 1
        
        return count + 1 if s[-1] != " " else count
    
    def countSegments(self, s: str) -> int:
        return len(s.split())